tests/sweep_framework/sweeps/all_gather_T3K_sharded.py

```diff
@@ -0,0 +1,272 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, Tuple
+
+import torch
+
+import ttnn
+from models.utility_functions import get_devices_for_t3000
+from tests.ttnn.utils_for_testing import check_with_pcc, start_measuring_time, stop_measuring_time
+from loguru import logger
+import tt_lib as ttl
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from tests.ttnn.unit_tests.operations.test_all_gather import is_unsupported_case
+from ttnn import ShardTensorToMesh
+
+# Override the default timeout in seconds for hang detection.
+TIMEOUT = 30
+
+# Parameters provided to the test vector generator are defined here.
+# They are defined as dict-type suites that contain the arguments to the run function as keys, and lists of possible inputs as values.
+# Each suite has a key name (in this case "suite_1" and "suite_2") which will associate the test vectors to this specific suite of inputs.
+# Developers can create their own generator functions and pass them to the parameters as inputs.
+
+input_shapes = []
+
+for batch_size in range(1, 10):  # Increment by 1
+    for channels in range(1, 10):  # Increment by 1
+        for height in range(32, 128, 32):  # Increment by 32
+            for width in range(32, 1024, 32):  # Increment by 32
+                input_shapes.append([batch_size, channels, height, width])
+
+parameters = {
+    "line_all_gather": {
+        "num_devices": [4, 8],
+        "num_links": [1, 2],
+        "input_shape": input_shapes,
+        "input_shard_shape": [
+            [32, 1024],
+            [32, 128],
+            [32, 32],
+            [128, 32],
+            [64, 64],
+            [64, 32],
+            [32, 64],
+            [128, 128],
+            [35, 35],
+            [41, 41],
```

Comment:
Padded shard shape? This will be in the padded shards bucket I think.

This reminds me... I'll need to add a sweep category for negative tests that are expected to fail.